### DISCUSSION TITLE
Building a 21st century light organ in a 1930s vintage radio housing

### DISCUSSION THREAD

**Discussion by dieterstoll** [hvianna/audioMotion.js#59] [Show and tell]
TL;DR: go to https://github.com/hvianna/audioMotion.js/discussions/48

I found a radio ruin on eBay. Someone had foolishly vandalised a 1930 Blaupunkt radio receiver (admittedly, not the top tier quality of radio receivers insofar as it wasn't a superset but an audion based receiver). Anyway they had ripped out the control panel, grafted a 1980s car radio onto it and drilled a hole in the top for an antenna. Loudspeaker was crap. I tore out all the electronics. Removing the loudspeaker and the cloth left a huge square opening. I found a 20cm diameter convex lens on eBay:
![Screenshot 2024-04-12 at 19 27 40](https://github.com/hvianna/audioMotion.js/assets/56186691/8377ccb6-9706-40a5-82b9-689f060e72e5)
So I 3D-printed a suitable mask for the lens (since my build plate on the Snapmaker 250 isn't large enough I divided it into four quarters). The lens sits a few centimetres in front of a 12.1" 4:3 display (1024x768 resolution): https://www.amazon.com/-/en/dp/B0CLXPWT8C/
and the display is connected via HDMI to a Raspberry Pi 4 with a USB microphone, running Bookworm. So far so good.
I wanted the whole shebang to be fully autonomous, with no need for WiFi connectivity or keyboard / mouse interaction. To this end I had to find ways to use audioMotion.js in a "sort of" headless mode. And thanks to Henrique's generous help I succeeded in running a web server on the Pi (to avoid having to authorise the use of the microphone), starting audio motion.js with all the parameters in place, and to simulate the "f" keypress that is required to turn on Full Screen Mode. The latter was something of a weird trip. Since Bookworm no longer uses X11/openbox as its window manager but Wayland. Wayland is sort of infamous for breaking things that used to work under X11. Such was the case here. I used raspi-config to revert to X11/openbox. Details here: https://github.com/hvianna/audioMotion.js/discussions/48
So here's a quick'n'dirty video:
https://youtu.be/75wIm6cu9sU

Final step: making the entire filesystem read-only. Reason: the Pimoroni on-off shim that I have used on about a dozen previous raspi projects no longer works on Bookworm. I'm pretty sure it's something trivial but after half a day of trial and error I gave up. I have now reverted to simply turning off power by a switch which shouldn't be a problem with a read-only filesystem.

**Comment by hvianna** [comment]
This is so cool! A portable spectrum analyzer in a real-life, physical gadget! And I love the retro-futuristic look!
I'm honored you decided to use audioMotion for your project! Cheers! 🍻
